filesystem/filesystem.go

```diff
@@ -581,7 +652,7 @@ func (m *Mount) AddLinkedProtector(descriptor string, dest *Mount) (bool, error)
 		}
 		return false, nil
 	}
-	if !os.IsNotExist(err) {
+	if errors.Cause(err) != unix.ENOENT {
```

Comment:
If we end up being forced to use raw errors (which we hopefully won't), we should have `readMetadataFileSafe` return an eror so that `IsNotExist` still works.

Reply:
Changed to use `os` instead of `unix` directly.